src/Features/CSharp/Portable/ReplaceDiscardDeclarationsWithAssignments/CSharpReplaceDiscardDeclarationsWithAssignmentsService.cs

```diff
@@ -21,21 +22,72 @@ namespace Microsoft.CodeAnalysis.CSharp.ReplaceDiscardDeclarationsWithAssignment
-                    localDeclarationStatement.Declaration.Variables.Any(IsDiscardDeclaration))
+                switch (child)
                 {
-                    RemoveDiscardHelper.ProcessDeclarationStatement(localDeclarationStatement, editor);
-                }
-                else if (child is CatchDeclarationSyntax catchDeclaration &&
-                    IsDiscardDeclaration(catchDeclaration))
-                {
-                    // "catch (Exception _)" => "catch (Exception)"
-                    editor.ReplaceNode(catchDeclaration, catchDeclaration.WithIdentifier(default));
+                    case LocalDeclarationStatementSyntax localDeclarationStatement:
+                        if (localDeclarationStatement.Declaration.Variables.Any(IsDiscardDeclaration))
+                        {
+                            RemoveDiscardHelper.ProcessDeclarationStatement(localDeclarationStatement, editor);
+                        }
+
+                        break;
+
+                    case CatchDeclarationSyntax catchDeclaration:
+                        if (IsDiscardDeclaration(catchDeclaration))
+                        {
+                            // "catch (Exception _)" => "catch (Exception)"
+                            editor.ReplaceNode(catchDeclaration, catchDeclaration.WithIdentifier(default));
+                        }
+
+                        break;
+
+                    case DeclarationExpressionSyntax declarationExpression:
+                        if (declarationExpression.Designation is DiscardDesignationSyntax discardSyntax)
+                        {
+                            // "M(out var _)" => "M(out _)"
+                            // "M(out int _)" => "M(out _)"
+
+                            var discardToken = SyntaxFactory.Identifier(
+                                leading: declarationExpression.GetLeadingTrivia(),
+                                contextualKind: SyntaxKind.UnderscoreToken,
+                                text: discardSyntax.UnderscoreToken.Text,
+                                valueText: discardSyntax.UnderscoreToken.ValueText,
+                                trailing: declarationExpression.GetTrailingTrivia());
+                            var replacementNode = SyntaxFactory.IdentifierName(discardToken);
+
+                            // Removing explicit type is possible only if there are no overloads of the method with same parameter.
+                            // For example, if method "M" had overloads with signature "void M(int x)" and "void M(char x)",
+                            // then the replacement "M(out int _)" => "M(out _)" will cause overload resolution error.
+                            // Bail out if replacement changes semantics.
+                            var speculationAnalyzer = new SpeculationAnalyzer(declarationExpression,
+                                replacementNode, semanticModel, cancellationToken, skipVerificationForReplacedNode: true);
```

Comment:
is ti correct to have `skipVerificationForReplacedNode: true`.  I've definitley had bugs at times because i didn't check the replaced node .  If it is correct, can you have a comment as for why?

Reply:
I wasn't sure if it mattered to check semantics for the argument itself, as we do check the semantics for outer invocation is unchanged. I can add a comment.

Reply:
Actually, `skipVerificationForReplacedNode = false` did not break anything, so I just removed the explicit argument.